### PR TITLE
Move `--version` logic into `cmd/--version.rb`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

`time brew --version` is the same before and after.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Makes sense, thanks @reitermarkus!

File: Library/Homebrew/cmd/--version.rb [hunk 1]
```diff
@@ -5,10 +5,9 @@ module Homebrew
   module_function
 
   def __version
-    # As a special case, `--version` is implemented directly in `brew.rb`. This
-    # file merely serves as a container for the documentation. It also catches
-    # the case where running `brew --version` with additional arguments would
-    # produce a rather cryptic message about a non-existent `--version` command.
-    raise UsageError
+    raise UsageError, "This command does not take arguments." unless ARGV.empty?
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/--version.rb]: Could maybe use the same form here as e.g. `brew pull` i.e. the new option parser stuff?

Comment (reitermarkus) [reply Library/Homebrew/cmd/--version.rb]: I think this is better, since a `UsageError` shows the command's help along with the error message.

Comment (MikeMcQuaid) [reply Library/Homebrew/cmd/--version.rb]: @reitermarkus I disagree; I'd rather see `Homebrew::CLI::Parser.parse` extended to output the command's help and have more consistent behaviour for now.

**Pull request merged**